File: Algorithms/Searching/Binary_Search/Binary_Search.py
# Returns index of x in arr if present, else -1
def binary_search(arr, low, high, x):

    # Check base case
    if high >= low:
        mid = (high + low) // 2

        # If element is present at the middle itself
        if arr[mid] == x:
            return mid

        # If element is smaller than mid, then it can only
        # be present in left subarray
        elif arr[mid] > x:
            return binary_search(arr, low, mid - 1, x)

        # Else the element can only be present in right subarray
        else:
            return binary_search(arr, mid + 1, high, x)

    # Element is not present in the array
    else:
        return -1


# Test
arr = [2, 13, 15, 70, 19, 4, 25, 6]
x = 13

# Function call
result = binary_search(arr, 0, len(arr) - 1, x)

if result != -1:
    print("Element is present at index", str(result))
else:
    print("Element is not present in array")

File: Algorithms/Searching/Binary_Search/BinarySearch.cs
using System;

namespace Algorithms
{
    internal class BinarySearch
    {
        /// <summary>
        /// Performs a binary search for an element in a sorted list.
        /// </summary>
        /// <param name="arr">Array from which to search</param>
        /// <param name="element">Element to search</param>
        /// <returns>Index of element in the list if present, else -1</returns>
        public static int search(int[] arr, int x)
        {
            var l = 0;
            var r = arr.Length - 1;

            // Keep searching from both ends
            while (l <= r)
            {
                int mid = l + (r - l) / 2;

                if (arr[mid] == x)
                {
                    return mid;
                }

                if (arr[mid] < x)
                {
                    l = mid + 1;
                }
                else
                {
                    r = mid - 1;
                }
            }

            return -1;
        }

        public static void Main(string[] args)
        {
            int[] arr = { 1, 2, 3, 5 };
            int size = arr.Length;
            int element = 3;
            
            Console.WriteLine("Position of {0} in the list [{1}] is {2}",
                element, string.Join(", ", arr), search(arr, element));

            Console.ReadKey();
        }
    }
}
